### PR TITLE
feat: sanitize vote description

### PR DESCRIPTION
**Pull request by PhearZero**
# Overview

- Sanitize Proposal Descriptions

### Issue

- https://algorandfoundation.atlassian.net/browse/AF-112

### CODE REVIEW

File: src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx [hunk 1]
```diff
@@ -67,6 +69,8 @@ function validateProposalCsv(value: string, ctx: z.RefinementCtx) {
       }
     })
   })
+
+  return Papa.unparse(parsed as unknown as UnparseObject<Proposal>)
```
Comment (ori-shem-tov) [inline src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx:73]:
```suggestion
return Papa.unparse({ data: parsed.data, fields: parsed.meta.fields || [] })
```
According to https://www.papaparse.com/docs#unparse, it expects an object with `data` and `fields` properties.
@PhearZero what do you think about the above suggestion?

File: src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx [hunk 2]
```diff
@@ -24,7 +25,7 @@ const formSchema = zfd.formData({
   start: zfd.text(),
   end: zfd.text(),
   communityGrantAllocation: zfd.numeric(z.number().positive().min(1, 'Must be at least 1')),
-  proposalFile: zfd.text(z.string().trim().min(1, 'Required').superRefine(validateProposalCsv)),
+  proposalFile: zfd.text(z.string().trim().min(1, 'Required').transform(validateProposalCsv)),
```
Comment (ori-shem-tov) [inline src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx:28]: What about changing `validateProposalCsv` to `validateAndSantizeProposalCsv`?

File: src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx (hunk 1 above)
Comment (PhearZero) [reply src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx:73]: The main issue I have with changing the pattern is the additional object created causing more work for the GC. The documentation shows passing the parsed object directly to the unparse method.

File: src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx (hunk 2 above)
Comment (PhearZero) [reply src/xgov-dapp/src/features/vote-creation/RoundInfo.tsx:28]: Absolutely!

**Review by ori-shem-tov — APPROVED** [review]

**Pull request merged**